core/src/main/scala/org/http4s/TransferCoding.scala

```diff
@@ -18,23 +18,52 @@
  */
 package org.http4s
 
-import org.http4s.syntax.string._
+import cats.{Order, Show}
+import cats.data.NonEmptyList
 import org.http4s.util._
+import org.http4s.parser.{Http4sParser, Rfc2616BasicRules}
 
-final case class TransferCoding private (coding: CaseInsensitiveString) extends Renderable {
+sealed abstract case class TransferCoding private (coding: String) extends Renderable {
   override def render(writer: Writer): writer.type = writer.append(coding.toString)
 }
 
-object TransferCoding extends Registry {
-  type Key = CaseInsensitiveString
-  type Value = TransferCoding
+object TransferCoding {
+  private class TransferCodingImpl(coding: String) extends TransferCoding(coding)
 
-  implicit def fromKey(k: CaseInsensitiveString): TransferCoding = new TransferCoding(k)
+  // https://www.iana.org/assignments/http-parameters/http-parameters.xml#transfer-coding
+  val chunked: TransferCoding = new TransferCodingImpl("chunked")
+  val compress: TransferCoding = new TransferCodingImpl("compress")
+  val deflate: TransferCoding = new TransferCodingImpl("deflate")
+  val gzip: TransferCoding = new TransferCodingImpl("gzip")
+  val identity: TransferCoding = new TransferCodingImpl("identity")
+
+  /**
+    * Parse a list of Transfer Coding entries
+    */
+  def parseList(s: String): ParseResult[NonEmptyList[TransferCoding]] =
+    new Http4sParser[NonEmptyList[TransferCoding]](s, "Invalid Transfer Coding")
+    with Rfc2616BasicRules {
+      private def codingRule = rule {
+        "chunked" ~ push(chunked) |
+          "compress" ~ push(compress) |
+          "deflate" ~ push(deflate) |
+          "gzip" ~ push(gzip) |
+          "identity" ~ push(identity)
+      }
+
+      def main = rule {
+        oneOrMore(codingRule).separatedBy(ListSep) ~> { codes: Seq[TransferCoding] =>
+          NonEmptyList.of(codes.head, codes.tail: _*)
+        }
+      }
+    }.parse
+
+  implicit val http4sInstancesForTransferCoding: Show[TransferCoding] with Order[TransferCoding] =
```

Comment:
`Scheme` also implements `HttpCodec`. Should I have it in `TransferCoding` as well?

Reply:
I think so.  That shows that any value's serialization can be parsed to an equal value.  It's a nice property.